src/tree/build-tree.cc

```diff
@@ -212,8 +214,26 @@ EventMap *BuildTree(Questions &qopts,
                                                               &num_removed);
     KALDI_LOG <<  "BuildTree: removed "<< num_removed << " leaves.";
 
-    int32 num_leaves = 0;
-    EventMap *tree_renumbered = RenumberEventMap(*tree_clustered, &num_leaves);
+    int32 num_leaves_out = 0;
+    EventMap *tree_renumbered;
+    if (round_num_leaves) {
+      int32 num_leaves_required = ((num_leaves - num_removed) / 8) * 8;
+      std::vector<EventMap*> leaf_mapping;
+
+      int32 num_actually_removed = ClusterEventMapGetMapping(
+          *tree_clustered, stats, -kLogZeroBaseFloat, &leaf_mapping, 
```

Comment:
Perhaps a more idiomatic way to get positive infinity is `std::numeric_limits<BaseFloat>::infinity()` instead of `-kLogZeroBaseFloat`

Reply:
I agree with Daniel on how to get infinity.  Is this tested?  And did you test that the regular tree-building (without this option set) runs the same way it did before?

Reply:
I made the changes. I tested it that it works in the normal tree building case. I also tested that it works the same without the option set.